### PR TITLE
Add Pebble implementation

### PR DESCRIPTION
**Pull request by neurodrone**
cc @harshita for review.

### CODE REVIEW

File: pebble/pathapp/wscript [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+#
+# This file is the default set of rules to compile a Pebble project.
+#
+# Feel free to customize this to your needs.
+#
+
+import os.path
+
+top = '.'
+out = 'build'
```
Comment (harshita) [inline pebble/pathapp/wscript:11]: Are these two vars being used anywhere?

File: pebble/pathapp/wscript [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    ctx.load('pebble_sdk')
+
+    build_worker = os.path.exists('worker_src')
+    binaries = []
+
+    for p in ctx.env.TARGET_PLATFORMS:
+        ctx.set_env(ctx.all_envs[p])
+        ctx.set_group(ctx.env.PLATFORM_NAME)
+        app_elf='{}/pebble-app.elf'.format(ctx.env.BUILD_DIR)
+        ctx.pbl_program(source=ctx.path.ant_glob('src/**/*.c'),
```
Comment (harshita) [inline pebble/pathapp/wscript:29]: Is it possible to declare the path here and on line 40 in some properties file or at an application level?

File: pebble/pathapp/src/pathapp.c [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+	NULL,
+};
+
+struct direction {
+	char *title;
+	char *stub;
+};
+
+static struct direction directions[] = {
+	{ .title = "JSQ to 33rd", .stub = "jsq_33rd" },
```
Comment (harshita) [inline pebble/pathapp/src/pathapp.c]: We discussed this offline. It would less confusing to just see destination instead of source-destination, since the user of the app needs to know only 2 things: where he/she is at and where he/she wants to go.

File: pebble/pathapp/src/pathapp.c [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+	app_message_outbox_send();
+
+	// Zero out the buffer area we used. We can leave the last
+	// NUL byte as it is, so we don't need to do strlen(buffer)+1 here.
+	memset(buffer, 0x00, strlen(buffer));
+}
+
+static void sched_menu_callback(int index, void *ctx) {
+	vibes_short_pulse();
+	sched_items[index].subtitle = "Ok. Now hurry!";
```
Comment (harshita) [inline pebble/pathapp/src/pathapp.c:81]: Can you declare all the app messages as static strings somewhere or even an enum for all messages is fine.

File: pebble/pathapp/src/pathapp.c [hunk 5]
```diff
@@ -0,0 +1,285 @@
+#include "pebble.h"
```
Comment (harshita) [inline pebble/pathapp/src/pathapp.c:1]: Can you add unit/integration tests for this and every app related file? I notice there are none!

File: pebble/pathapp/src/pathapp.c [hunk 6]
```diff
@@ -0,0 +1,285 @@
...
+static void send_to_phone(int key, const char *direction) {
+	DictionaryIterator *iter;
+	char *buf;
+
+	app_message_outbox_begin(&iter);
+	if (!iter) {
+		return;
+	}
+
+	buf = buffer;
```
Comment (harshita) [inline pebble/pathapp/src/pathapp.c]: Can you rename buf to something more intuitive, like `bufptr`?

File: pebble/pathapp/src/pathapp.c [hunk 7]
```diff
@@ -0,0 +1,285 @@
...
+	window_layer = window_get_root_layer(window);
+	bounds = layer_get_frame(window_layer);
+
+	sched_layer = simple_menu_layer_create(bounds, window, sched_sections, 1, NULL);
+
+	layer_add_child(window_layer, simple_menu_layer_get_layer(sched_layer));
+}
+
+static void schedule_window_unload(Window *window) {
+	memset(buffer, 0x00, sizeof buffer);
```
Comment (harshita) [inline pebble/pathapp/src/pathapp.c:211]: Can you add comments on what is intended here?

Comment (neurodrone) [inline pebble/pathapp/src/pathapp.c:211]: Addressed via https://github.com/neurodrone/path/commit/fdfc472bdc341578c721ef77d59b782f220f84a2.

File: pebble/pathapp/src/pathapp.c (hunk 6 above)
Comment (neurodrone) [inline pebble/pathapp/src/pathapp.c]: Renamed via https://github.com/neurodrone/path/commit/6f066bdb0efed3e58b6c63bed6ec6221399eb76c.

File: pebble/pathapp/src/pathapp.c (hunk 3 above)
Comment (neurodrone) [inline pebble/pathapp/src/pathapp.c]: Addressed via https://github.com/neurodrone/path/commit/f148cc34e07bd606ce67c441498e55f0042924ad.

File: pebble/pathapp/wscript (hunk 1 above)
Comment (neurodrone) [inline pebble/pathapp/wscript:11]: They are used within the pebble [Waf builder](https://waf.io/apidocs/index.html) to denote the names for top-level and build directories respectively I guess.

File: pebble/pathapp/wscript (hunk 2 above)
Comment (neurodrone) [inline pebble/pathapp/wscript:29]: `pebble --new-project` autogens this file, so I think we can leave this part as it is.

File: pebble/pathapp/src/pathapp.c (hunk 4 above)
Comment (neurodrone) [inline pebble/pathapp/src/pathapp.c:81]: I can make this a constant, but in C that would still be a static NUL-terminated char array somewhere whose pointer I will pass here. Is that fine?

File: pebble/pathapp/src/pathapp.c (hunk 5 above)
Comment (neurodrone) [inline pebble/pathapp/src/pathapp.c:1]: Don't see an easy way to perform integration tests for pebble in C or mock the message-passing or Layer-based UI generation yet. I can look into it. Is it fine if I make an issue now and address this in a different PR?

Comment (harshita) [inline pebble/pathapp/src/pathapp.c:1]: That would be reasonable.

Comment (neurodrone) [inline pebble/pathapp/src/pathapp.c:1]: Create an issue: https://github.com/neurodrone/path/issues/2.

**Comment by harshita** [comment]
:+1:

**Comment by neurodrone** [comment]
:sunglasses:
